src/sql/src/normalize.rs

```diff
@@ -108,16 +137,34 @@ pub fn option_objects(options: &[SqlOption<Aug>]) -> BTreeMap<String, SqlOption<
 /// Unnormalizes an object name.
 ///
 /// This is the inverse of the [`unresolved_object_name`] function.
-pub fn unresolve(name: FullName) -> UnresolvedObjectName {
+pub fn unresolve(name: FullObjectName) -> UnresolvedObjectName {
     let mut out = vec![];
-    if let DatabaseSpecifier::Name(n) = name.database {
+    if let RawDatabaseSpecifier::Name(n) = name.database {
         out.push(Ident::new(n));
     }
     out.push(Ident::new(name.schema));
     out.push(Ident::new(name.item));
     UnresolvedObjectName(out)
 }
 
+/// Converts an `UnresolvedObjectName` to a `FullObjectName` if the
+/// `UnresolvedObjectName` is fully specified. Otherwise returns an error.
+pub fn full_name(raw_name: &mut UnresolvedObjectName) -> Result<FullObjectName, anyhow::Error> {
```

Comment:
hi! Little bit of a driveby but I just did a rebase over this change and now I need to generate a `FullObjectName` with this function (or so I think -- I was planning to tag you in the PR since I'm probably not doing it the right way tm)

anyway, my question: why does this take a `&mut UnresolvedObjectName` rather than a `mut UnresolvedObjectName`? I was a tad confused how to use this function correctly because I didn't know if the leftover `raw_name` would still have some meaning but, as I mentioned, I'm out of my depth here in sql land so wanted to check in!

Reply:
Looking at it now, there's really no good reason that it takes `&mut UnresolvedObjectName` instead of `mut UnresolvedObjectName`. In fact it probably should take `mut UnresolvedObjectName` since it ends up popping everything out of the `raw_name`.

Feel free to change that or if you want I can change it.

Though in general this method is really only meant for `UnresolvedObjectName`s that you know are fully qualified, which doesn't happen very often.

Reply:
If you want to look up a name that you don't know if it's fully qualified you'll want to do something like this
```
let name = UnresolvedObjectName {...};
let name = normalize::unresolved_object_name(name);
some_reference_to_a_session_catalog_object.resolve_item(&name);
```

Reply:
Right! That's what I do at first but then I need a `FullObjectName` to call `CatalogItem::desc` on what's returned from `resolve_item` (I think this is a pretty recent change as I only had to fix it in a rebase this morning)

Reply:
Oy, I really tried to avoid that param in `desc` because it's really only for a clean error message, but I couldn't figure out how. If you take a look at other usages of `desc` you can see how to get the full object name.

It should look something like
```rust
let item = CatalogItem{...}
let name = some_reference_to_session_catalog.resolve_full_name(item.name());
let desc = item.desc(name)?;
```

Reply:
If you don't have a session catalog but only have a normal catalog then it looks something like:
```
let name = some_reference_to_catalog.resolve_full_name(item.name(), item.conn_id().unwrap_or(SYSTEM_CONN_ID));

Reply:
Thanks for the tip! I've updated #11306 to use `resolve_full_name`